src/common/src/DocumentViewer/DocumentViewer.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useState } from 'react';
+import { Alert, Heading, themeColor } from '@amsterdam/asc-ui';
+import { DocumentViewerStyle, SpinnerStyle } from './DocumentViewerStyles';
+import DocumentRenderer from './DocumentRenderer';
+
+type Props = {
+	uri: string;
+	authorizationHeader?: string;
+	onFailure?: (e: Error) => void;
+};
+
+export type DocumentState = {
+	loading: boolean;
+	filename?: string;
+	error?: string;
+	contentType?: string;
+};
+
+const DocumentViewer: React.FC<Props> = ({ uri, authorizationHeader, onFailure }: Props) => {
+	const [documentState, setDocumentState] = useState<DocumentState>({ loading: true });
+	const { loading, filename, error, contentType } = documentState;
+
+	const updateDocumentState = (documentState: Partial<DocumentState>) => {
+		setDocumentState((prevState) => ({
+			...prevState,
+			...documentState,
+		}));
+	};
+
+	const getRequestHeaders = (): HeadersInit => (authorizationHeader ? { Authorization: authorizationHeader } : {});
+
+	useEffect(() => {
+		const filename = uri.split('/').pop() || 'Onbekend bestand';
+		updateDocumentState({ loading: true, error: undefined, contentType: undefined, filename });
+
+		fetch(uri, { method: 'HEAD', headers: getRequestHeaders() })
+			.then((response) => {
+				if (!response.ok)
+					throw new Error(response.status === 404 ? 'Document niet gevonden.' : 'Fout bij het ophalen.');
+
+				return response.headers;
+			})
+			.then((headers) => {
```

Comment:
I don't think this `.then()` adds any value. There is no promise being returned from the preceding `.then()`.

Reply:
Removed

Reply:
:ok_hand: